oncall/autoIssueAssign.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+autoAssign() {
+  curl -X POST "https://api.github.com/repos/microsoft/Oryx/issues/$3/assignees" -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $2" -d "{\"assignees\": ["\"$1\""]}"
+}
+
+checkCustomOncall() {
+  input=$1
+  today=$2
+  token=$3
+  issue=$4
+  currentOncall=$5
+  if [ -z "$input" ]; then
+    return 1
+  fi
+
+  IFS="," read -ra arr <<< "$1"
+  name=${arr[0]}
+  startDate=${arr[1]}
+  endDate=${arr[2]}
+  startDate=$(date -d $startDate +%s)
+  endDate=$(date -d $endDate +%s)
+  if [ "$today" -ge "$startDate" ] && [ "$today" -le "$endDate" ]; then
+    echo "Custom shift found"
+    autoAssign $name $token $issue
+    return 0
+  else
+    autoAssign $currentOncall $token $issue
+    return 0
+  fi
+}
+
+# https://medium.com/@Drew_Stokes/bash-argument-parsing-54f3b81a6a8f
+ONCALLS=""
+while (( "$#" )); do
+  case "$1" in
+    --token) #Github token flag
+    authToken=$2
+    shift 2
+    ;;
+    --issue) # Issue number flag
+    issueNum=$2
+    shift 2
+    ;;
+    --custom) # Custom shift flag
```

Comment:
Hey, so in the demo video there was a custom shift flag. Did you decide to do away with that in favor of a single specified custom shift?

If so, I would lean towards going back to the file approach and parsing through every line of the file, as cormac suggests. This would allow us to plan a few custom shifts in advance, especially if they are back-to-back without having any gap time